samples/11.qnamaker/src/main/java/com/microsoft/bot/sample/qnamaker/Application.java

```diff
@@ -1,2 +1,78 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+
+package com.microsoft.bot.sample.qnamaker;
+
+import com.microsoft.bot.builder.Bot;
+import com.microsoft.bot.integration.AdapterWithErrorHandler;
+import com.microsoft.bot.integration.BotFrameworkHttpAdapter;
+import com.microsoft.bot.integration.CredentialProvider;
+import com.microsoft.bot.integration.ConfigurationCredentialProvider;
+import com.microsoft.bot.integration.Configuration;
+import com.microsoft.bot.integration.spring.BotController;
+import com.microsoft.bot.integration.spring.BotDependencyConfiguration;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+
+//
+// This is the starting point of the Sprint Boot Bot application.
+//
+@SpringBootApplication
+
+// Use the default BotController to receive incoming Channel messages. A custom
+// controller could be used by eliminating this import and creating a new
+// org.springframework.web.bind.annotation.RestController.
+// The default controller is created by the Spring Boot container using
+// dependency injection. The default route is /api/messages.
+@Import({BotController.class})
+
+/**
+ * This class extends the BotDependencyConfiguration which provides the default
+ * implementations for a Bot application.  The Application class should
+ * override methods in order to provide custom implementations.
+ */
+public class Application extends BotDependencyConfiguration {
+    public static void main(String[] args) {
+        SpringApplication.run(Application.class, args);
+    }
+
+    /**
+     * Returns the Bot for this application.
+     *
+     * <p>
+     *     The @Component annotation could be used on the Bot class instead of this method
+     *     with the @Bean annotation.
+     * </p>
+     *
+     * @return The Bot implementation for this application.
+     */
+    @Bean
+    public Bot getBot() {
+        return new QnABot();
+    }
+
+    /**
+     * Returns a custom Adapter that provides error handling.
+     *
+     * @param configuration The Configuration object to use.
+     * @return An error handling BotFrameworkHttpAdapter.
+     */
+    @Override
+    public BotFrameworkHttpAdapter getBotFrameworkHttpAdaptor(Configuration configuration) {
+        return new AdapterWithErrorHandler(configuration);
+    }
+    
+    /**
+     * Returns a credential provider to be used with the Bot Framework Adapter.
+     *
+     * @param configuration The Configuration object to use.
+     * @return A credential provider ConfigurationCredentialProvider.
+     */
+    @Override
+    public CredentialProvider getCredentialProvider(Configuration configuration) {
+        return new ConfigurationCredentialProvider(configuration);
+    }
```

Comment:
After reviewing the [BotDependencyConfiguration](https://github.com/southworks/botbuilder-java/blob/main/libraries/bot-integration-spring/src/main/java/com/microsoft/bot/integration/spring/BotDependencyConfiguration.java), I think that this can be removed as this is overriding with the same behavior the one of the [base](https://github.com/southworks/botbuilder-java/blob/main/libraries/bot-integration-spring/src/main/java/com/microsoft/bot/integration/spring/BotDependencyConfiguration.java#L79).

So, we can ramove this :)!

Reply:
Done